src/expr/kind_template.h

```diff
@@ -137,39 +96,9 @@ inline TheoryId& operator ++ (TheoryId& id) {
   return id = static_cast<TheoryId>(((int)id) + 1);
 }
 
-inline std::ostream& operator<<(std::ostream& out, TheoryId theoryId) {
-  switch(theoryId) {
-${theory_descriptions}
-#line 144 "${template}"
-  default:
-    out << "UNKNOWN_THEORY";
-    break;
-  }
-  return out;
-}
-
-inline TheoryId kindToTheoryId(::CVC4::Kind k) {
-  switch(k) {
-  case kind::UNDEFINED_KIND:
-  case kind::NULL_EXPR:
-    break;
-${kind_to_theory_id}
-#line 158 "${template}"
-  case kind::LAST_KIND:
-    break;
-  }
-  throw IllegalArgumentException("", "k", __PRETTY_FUNCTION__, "bad kind");
-}
-
-inline TheoryId typeConstantToTheoryId(::CVC4::TypeConstant typeConstant) {
-  switch(typeConstant) {
-${type_constant_to_theory_id}
-#line 168 "${template}"
-  case LAST_TYPE:
-    break;
-  }
-  throw IllegalArgumentException("", "k", __PRETTY_FUNCTION__, "bad type constant");
-}
+std::ostream& operator<<(std::ostream& out, TheoryId theoryId);
+TheoryId kindToTheoryId(::CVC4::Kind k) CVC4_PUBLIC;
```

Comment:
I am a bit worried kindToTheoryId and typeConstantToTheoryId might have a noticeable performance impact on larger scale experiments. (Say 1-2% maybe? Though that sounds high.) Has a performance measurement been done for this change?